packages/discord.js-utilities/src/lib/MessagePrompter/MessagePrompter.ts

```diff
@@ -0,0 +1,132 @@
+import { MessagePrompterReactionStrategy } from './strategies/MessagePrompterReactionStrategy';
+import type {
+	IMessagePrompterConfirmStrategyOptions,
+	IMessagePrompterNumberStrategyOptions,
+	IMessagePrompterReactionStrategyOptions
+} from './strategyOptions';
+
+/**
+ * This is a [[MessagePrompter]], a utility that sends a message, prompting for user input. The prompt can resolve to any kind of input.
+ * There are several specifiable types to prompt for user input, and they are as follows:
+ * - Confirm
+ *   This will send a simple Yes/No prompt, using reactions.
+ * - Number
+ *   This will prompt for an integer. By default it will be a number between 0 and 10 (inclusive), however you can also specify your own custom range (inclusive).
+ * - Reactions
+ *   This can be any kind of reaction emoji that Discord supports, and as many as you want. This type will return that reaction instead of a boolean.
+ * - Message
+ *   This will prompt the user and require a response in the form of a message. This can be helpful if you require a user to upload an image for example, or give text input.
+ *
+ * You must either use this class directly or extend it.
+ *
+ * [[MessagePrompter]] uses reactions to prompt for a yes/no answer and returns it.
+ * You can modify the confirm and cancel reaction used for each message, or use the [[MessagePrompter.defaultPrompts]].
+ * [[MessagePrompter.defaultPrompts]] is also static so you can modify these directly.
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you sure you want to continue?');
+ * const result = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Choose a number between 5 and 10?', 'number', {
+ * 		start: 5,
+ * 		end: 10
+ * });
+ * const result = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you happy or sad?', 'reaction', {
+ * 		reactions: ['🙂', '🙁']
+ * });
+ * const result = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Do you love me?', 'message');
+ * const result = await handler.run(channel, author);
+ * ```
+ */
+export class MessagePrompter {
+	/**
+	 * The strategy used in [[MessagePrompter.run]]
+	 */
+	public strategy: MessagePrompterBaseStrategy;
+
+	/**
+	 * Constructor for the [[MessagePrompter]] class
+	 * @param message The message to send.
+	 * @param strategy The strategy name or Instance to use
+	 * @param strategyOptions The options that are passed to the strategy
+	 */
+	public constructor(
+		message: MessagePrompterMessage | MessagePrompterBaseStrategy,
+		strategy?: MessagePrompterStrategies,
+		strategyOptions?:
+			| IMessagePrompterConfirmStrategyOptions
+			| IMessagePrompterNumberStrategyOptions
+			| MessagePrompterReactionStrategy
```

Comment:
Seems like this is supposed to be `IMessagePrompterMessageStrategyOptions` ?

Reply:
`IMessagePrompterMessageStrategyOptions ` doesn't even exist, so I assume this line can be removed all together